src/sema/types.rs

```diff
@@ -220,24 +219,26 @@ pub fn resolve_fields(delay: ResolveFields, file_no: usize, ns: &mut Namespace)
             let def = ns.structs[struct_no].clone();
```

Comment:
Inlining function definitions makes the code unreadable.

Reply:
https://github.com/hyperledger-labs/solang/blob/779b4b4aab4d307943e7ac95452800758f8eb155/src/sema/types.rs#L219-L245

I think we can inline definition or use lambda functions when they're small sized pieces of code. Here, you defined a large function inside another one.

Reply:
Right, I was totally confused because you are referring to some existing code.

Reply:
You are right of course, that is not the most readable code. I've moved the function, looks better now (I think)